hw/system/occamy/sw/CMakeLists.txt

```diff
@@ -1,22 +1,135 @@
+function(add_cva6_executable name srcs)
+	# Aliases
+  set(common_srcs src/start.S)
+	# Create an executable linked against the cva6Runtime
+  # and dump its contents
+  add_executable(${name} ${common_srcs} ${srcs})
+	target_link_libraries(${name} PUBLIC cva6Runtime)
+	target_include_directories(${name} PUBLIC cva6Runtime/ include/)
+  dump_executable(${name})
+endfunction()
+
+# Creates an executable with program code for both CVA6 and Snitch cores.
+# Snitch code is additionally linked against snRuntime library.
+# You can pass additional options to the Snitch binary compilation
+# command, e.g. -I flags, via the sn_c_flags argument.
+function(add_hybrid_executable name srcs sn_srcs sn_c_flags)
+  # Aliases
+	set(SNITCH_ELF       ${SNITCH_PREFIX}_${name}.elf)
+	set(SNITCH_BIN       ${SNITCH_PREFIX}_${name}.bin)
+	set(SNITCH_TARGET    ${SNITCH_PREFIX}_${name})
+	set(SN_INCLUDE_FLAGS -I${SNITCH_RUNTIME_DIR}/include
+                       -I${SNITCH_RUNTIME_DIR}/vendor
+                       -I${SNITCH_RUNTIME_DIR}/vendor/riscv-opcodes
+                       -I${CMAKE_CURRENT_SOURCE_DIR}/include)
+	set(sn_srcs_ext      ${SNITCH_SOURCE_DIR}/start.S ${sn_srcs})
+	# Compile Snitch ELF
+	add_custom_command(
+    OUTPUT ${SNITCH_ELF}
+    COMMAND ${SN_C_COMPILER} ${SN_C_FLAGS} ${sn_c_flags} ${SN_INCLUDE_FLAGS} ${sn_srcs_ext} ${SN_LD_FLAGS} -o ${SNITCH_ELF}
+    DEPENDS ${sn_srcs_ext}
+  )
+  # Copy the ELF contents to a binary and to a human-readable dump file
+	add_custom_command(
+    OUTPUT ${SNITCH_BIN}
+    COMMAND ${SN_OBJCOPY} ${SN_OJBCOPY_FLAGS} ${SNITCH_ELF} ${SNITCH_BIN}
+    DEPENDS ${SNITCH_ELF}
+  )
+  dump_executable(${SNITCH_ELF})
+  # Add a target to build the Snitch binary
+	add_custom_target(
+    ${SNITCH_TARGET}
+    DEPENDS ${SNITCH_BIN}
+  )
+	add_dependencies(${SNITCH_TARGET} sn_runtime_lib_target)
+
+	# Hybrid target is a CVA6 executable with a dependency
+  # on the previously created Snitch binary
+	add_cva6_executable(${name} ${srcs})
+	add_dependencies(${name} ${SNITCH_TARGET})
+	target_compile_definitions(${name} PUBLIC -DSNITCH_BINARY="${SNITCH_BIN}")
+	target_include_directories(${name} PUBLIC ${SNITCH_BINARY_DIR})
+	target_include_directories(${name} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+endfunction()
+
+######################
+# Executable targets #
+######################
+
+add_cva6_executable(sanity src/sanity.c)
+add_cva6_executable(test_cva6_hello_world src/test_cva6_hello_world.c)
+add_cva6_executable(test_isolate src/test_isolate.c)
+add_cva6_executable(test_atomics src/test_atomics.c)
+add_cva6_executable(test_fence_t src/test_fence_t.c)
+add_hybrid_executable(test_tlbs "src/test_tlbs.c" "${SNITCH_SOURCE_DIR}/test_tlbs.c" "")
+add_hybrid_executable(test_atomics_multicore "src/test_atomics_multicore.c" "${SNITCH_SOURCE_DIR}/test_atomics_multicore.c" "")
+add_hybrid_executable(test_snitch_hello_world "src/test_snitch_hello_world.c" "${SNITCH_SOURCE_DIR}/test_snitch_hello_world.c" "")
```

Comment:
As mentioned already, I think the whole SW structure needs a rework and might get separated from the snitch repository. Let's discuss this in the meeting which we will soon hold for this topic :)